docs/development/workflow/development_workflow.rst

```diff
@@ -460,17 +478,49 @@ You are more likely to run into *conflicts* here--places where the changes you
 made conflict with changes that someone else made--than anywhere else. Ask for
 help if you need it.
 
-After the rebase you need to push your changes to GitHub; you will need force
-the push because `git`_ objects to re-writing the history of the repository
-after you have pushed it somewhere::
+... _howto_squash:
+
+How to squash
+=============
+
+Typically we ask to *squash*/*squish* when there was a fair amount of trial
+and error, but the final patch remains quite small, or when files were added
+and removed (especially binary files or files that should not remain in the
+repository) or if the number of commits in the history is disproportionate
+compared to the work being carried out (for example 30 commits gradually
+refining a final 10-line change).  Conceptually this is equivalent to
+exporting the final diff from a feature branch, then starting a new branch and
+only applying only that patch:
+
+The actual squashing is done on a new branch, using `git merge --squash`::
+
+  git checkout master -b squashed-new-feature
+  git merge --squash --no-commit my-new-feature
+  git commit -m "example of preferred squash message format"
+
+The new squashed branch can be checked and then renamed over the old one::
+
+  git branch -m my-new-feature tmp
+  git branch -m squashed-new-feature my-new-feature
+
+... _howto_push_force:
+
+How to push
+===========
+
+After using `git rebase` or `git merge --squash` you will still need
+to push your changes to GitHub so that they are visible to others and
+the Pull Request can be updated.  Use of a simple `git push` will be
+prevented because of the changed history, and will need to be manually
+overridden using::
 
-    git push -f
+    git push --force
```

Comment:
My preference is `-f` but if you like the verbosity I'm not opposed to `--force`.